### PR TITLE
Rebase tests with new fonttools

### PR DESCRIPTION
**Pull request by Tolker-KU**
Bumps fonttools to newest version and regenerate reference files for failing tests.

**Checklist**:

- [x] The GitHub pipeline is OK (green), 
      meaning that both `pylint` (static code analyzer) and `black` (code formatter) are happy with the changes of this PR.

- [x] A unit test is covering the code added / modified by this PR

- [x] This PR is ready to be merged 

- [x] In case of a new feature, docstrings have been added, with also some documentation in the `docs/` folder

- [x] A mention of the change is present in `CHANGELOG.md`

By submitting this pull request, I confirm that my contribution is made under the terms of the [GNU LGPL 3.0 license](https://github.com/PyFPDF/fpdf2/blob/master/LICENSE).

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -43,7 +43,7 @@ install_requires =
   Pillow>=6.2.2,!=9.2.*  # minimum version tested there: https://github.com/PyFPDF/fpdf2/actions/runs/2295868575
   # Version 9.2.0 is excluded due to DoS vulnerability with TIFF images: https://github.com/PyFPDF/fpdf2/issues/628
   # Version exclusion explained here: https://devpress.csdn.net/python/630462c0c67703293080c302.html
-  fonttools>=4.34.0
```
Comment (Lucas-C) [inline setup.cfg:46]: I agree that we should support fonttools>=4.41.1,
but why should we set this as the minimal version?

`fpdf2` can be used with older versions of `fonttools`,
so this change will restrain `fpdf2` compatibility with `fonttools`,
which could be annoying for end-users that need to use `fpdf2` with older versions of `fonttools`.

I'd prefer to only define there in `setup.cfg` the very absolute minimal versions of our dependencies that `fpdf2` can work with.

Also, do you know what change in a `fonttools` release generated those differences in reference PDF files?
I saw this in [the release notes for v4.39.0](https://github.com/fonttools/fonttools/releases/tag/4.39.0):

> [feaLib] Fixed handling of ignore statements with unmarked glyphs to match makeotf behavior, which assumes the first glyph is marked

**Comment by andersonhc** [comment]
I tested all the latest fonttools versions against our current reference files:
4.38.0 = OK
4.39.0 = OK
4.39.2 = OK
4.39.3 = OK
4.39.4 = OK
4.40.0 = OK
4.41.0 = Fail
4.41.1 = Fail

All the versions are producing valid and visually similar PDF files so we can say fpdf2 is compatible with all those versions, however some detail (like a timestamp) changed after fonttools 4.41.0 that is causing pytest to fail.

**Comment by Tolker-KU** [comment]
I've done some digging. Bytestreams for fonts change from version 4.40.0 to 4.41.0. At the moment I suspect that it might be this commit https://github.com/fonttools/fonttools/commit/4660b8c3598544a1ab650a58f7d0fb5d33714efc, but I'm not quite sure.

So our problem is that fonttools produces different bytestreams from different version even though PDFs probably always will be render the same way - at least to the human eye. Also the problem is only seen with a few fonts. So in order to have tests pass reference pdfs and the fonttools version must be in sync. This can be achieved either by pinning fonttools (and regenerating reference files if pinning to the new version) or by have different reference PDFs for different versions of fonttols, but that is probably not something we want.

File: setup.cfg (hunk 1 above)
Comment (Tolker-KU) [reply setup.cfg:46]: Some thoughts here: https://github.com/PyFPDF/fpdf2/pull/863#issuecomment-1648688622

**Comment by Lucas-C** [comment]
> This can be achieved either by pinning fonttools (and regenerating reference files if pinning to the new version) or by having different reference PDFs for different versions of fonttols, but that is probably not something we want.

Actually we could update the relevant PDF reference files, as you did in this PR, **WITHOUT** altering the pin on `fonttools` version in `setup.cfg`. And that is the option I would prefer to adopt 😊 

Could you please just try to update this PR and revert the change you made in `setup.cfg`? I think the pipeline should still pass.

The latest pipeline execution on `master` [has just failed due to the same 4 tests](https://github.com/PyFPDF/fpdf2/actions/runs/5650250481/job/15306270796):
* `test/fonts/test_font_fallback.py::test_fallback_font_with_overriden_get_fallback_font`
* `test/fonts/test_charmap.py::test_charmap_first_999_chars[Quicksand-Regular.otf]`
* `test/fonts/test_add_font.py::test_add_font_otf`
* `test/encryption/test_encryption.py::test_encrypt_font`

**Comment by Tolker-KU** [comment]
I've reverted the changes to setup.cfg and CHANGELOG.md. It works for users, but developers must be aware that tests will fail if they are running a too old version of fonttools

**Comment by Lucas-C** [comment]
> I've reverted the changes to setup.cfg and CHANGELOG.md. It works for users, but developers must be aware that tests will fail if they are running a too old version of fonttools

Agreed, perfect! 👍 

If that really bothers you, and you think that's useful, you can still add `fonttools>=4.41.1` to `test/requirements.txt`

**Comment by Tolker-KU** [comment]
I'm fine with the changes as is. We've this thread for reference if the problem ever comes up :)

**Comment by Lucas-C** [comment]
> I'm fine with the changes as is. We've this thread for reference if the problem ever comes up :)

Alright!

Then I'll merge this tomorrow in my time zone, when the pipeline will be finished and ✅

**Comment by Tolker-KU** [comment]
Python 3.7 got a too old version of fonttools making the tests fail :( https://github.com/PyFPDF/fpdf2/actions/runs/5650397495/job/15306727703?pr=863#step:6:114

Tests passed on Python 3.11

**Comment by andersonhc** [comment]
> Python 3.7 got a too old version of fonttools making the tests fail :( https://github.com/PyFPDF/fpdf2/actions/runs/5650397495/job/15306727703?pr=863#step:6:114
> 
> Tests passed on Python 3.11

Fonttools dropped support to python 3.7, that's why it's not downloading the new version on the 3.7 ci

**Comment by Lucas-C** [comment]
> Fonttools dropped support to python 3.7, that's why it's not downloading the new version on the 3.7 ci

Damn. You're right, it's mentioned in [the release notes for v4.39.0](https://github.com/fonttools/fonttools/releases/tag/4.39.0).

What are our options?
* also drop support for Python 3.7
* pin `fonttools==4.38.0`
* annotate those 4 unit tests so that they do not run under Python 3.7

I'd say the 3rd option is probably the best.
We already use some `skipif` logic: https://github.com/PyFPDF/fpdf2/blob/master/test/image/image_types/test_insert_images.py#L24

**Comment by Tolker-KU** [comment]
I've implemented the 3rd option. But we might wan't to consider dropping support for Python 3.7 soon-ish. 3.7 is EOL and will not receive security updates anymore. https://peps.python.org/pep-0537/#lifespan

File: test/requirements.txt [hunk 2]
```diff
@@ -4,7 +4,6 @@ black
 camelot-py
 endesive
 ghostscript  # required by camelot-py[base]: https://github.com/camelot-dev/camelot/blob/master/setup.py#L27
-fonttools>=4.41.1; python_version>'3.7' # can be removed when minimum version in setup.cfg surpasses this: https://github.com/PyFPDF/fpdf2/pull/863
```
Comment (Lucas-C) [inline test/requirements.txt]: 👍 Perfect

**Comment by Lucas-C** [comment]
Thank you for your time and for this contribution to `fpdf2` @Tolker-KU!

**Pull request merged**